### PR TITLE
Handle PaymentGatewayError

### PR DESCRIPTION
**Pull request by shreyas-satish**
This change adds a error handler for `APIError`; the handler sends an email to the `ADMINS` of the app with the configured.

### CODE REVIEW

File: boxoffice/views/custom_exceptions.py [hunk 1]
```diff
@@ -1,9 +1,24 @@
...
+from flask.ext.mail import Message
+from flask import jsonify, make_response
+from utils import cors
+
 
 class APIError(Exception):
 
-    def __init__(self, message, status_code):
+    def __init__(self, message, status_code, response_message):
         Exception.__init__(self)
```
Comment (jace) [inline boxoffice/views/custom_exceptions.py]: Use `super` as a general rule.

File: boxoffice/views/custom_exceptions.py [hunk 2]
```diff
@@ -1,9 +1,24 @@
...
         self.message = message
         self.status_code = status_code
+        self.response_message = response_message
+
+
+@app.errorhandler(APIError)
+@cors
+def handle_api_error(error):
+    msg = Message(subject='API Error', recipients=app.config['ADMINS'])
```
Comment (jace) [inline boxoffice/views/custom_exceptions.py]: Include Boxoffice here?

File: boxoffice/views/custom_exceptions.py [hunk 3]
```diff
@@ -1,9 +1,24 @@
...
+    msg.body = error.message
+    mail.send(msg)
```
Comment (jace) [inline boxoffice/views/custom_exceptions.py]: Sending mail from the error handler is a bit of a hack. This needs to go into a proper exception logging system.

File: boxoffice/views/order.py [hunk 4]
```diff
@@ -264,7 +264,8 @@ def payment(order):
         online_payment.fail()
         db.session.add(online_payment)
         db.session.commit()
-        raise APIError("Online payment failed for order - {order}".format(order=order.id), 502)
+        raise APIError("Online payment failed for order - {order} with the following details - {msg}".format(order=order.id,
+            msg=rp_resp.content), 500, 'Your payment failed. Please try again or contact us at {email}.'.format(email=order.organization.contact_email))
```
Comment (jace) [inline boxoffice/views/order.py]: 500 is internal server error. This is not an internal server error.

File: boxoffice/views/order.py [hunk 5]
```diff
@@ -333,7 +334,9 @@ def cancel_line_item(line_item):
                 online_payment=payment, amount=line_item.final_amount, currency=CURRENCY.INR))
             db.session.commit()
         else:
-            raise APIError("Cancellation failed for order - {order}".format(order=line_item.order.id), 502)
+            raise APIError("Cancellation failed for order - {order} with the following details - {msg}".format(order=order.id,
+                msg=rp_resp.content), 500,
```
Comment (jace) [inline boxoffice/views/order.py]: Use something other than 500.

File: boxoffice/views/custom_exceptions.py (hunk 2 above)
Comment (shreyas-satish) [inline boxoffice/views/custom_exceptions.py]: The Boxoffice email address in recipients?

Comment (shreyas-satish) [inline boxoffice/views/custom_exceptions.py]: The Boxoffice email address in recipients?

File: boxoffice/views/order.py (hunk 4 above)
Comment (shreyas-satish) [inline boxoffice/views/order.py]: Is 502 more apt?

File: boxoffice/views/custom_exceptions.py (hunk 3 above)
Comment (shreyas-satish) [inline boxoffice/views/custom_exceptions.py]: Sure, I'll write to a log file. But, it's also useful for us to know when something like this occurs, right? For which, I thought email works well.

**Pull request merged**